### PR TITLE
Events: persist + display keyword tally

### LINKED ISSUE #92: Cleanup: better error handling
see https://github.com/Sotera/watchman/commit/1ff03364b5b48f06710a740f981f19efc54991db

### LINKED ISSUE #119: Remove Word cloud
Have the event visualization reflect what we are storing in mongo/sending to uncharted more accurately by using the keywords supplied in Dr. M.

### PR DESCRIPTION
**Pull request by lukewendling**
fix #119, fix #92 

## Tasks

- [x] replace word cloud with simple tabular view
- [x] save keyword tallies to event store
- [x] update Loopy with static methods for convenience helpers
- [x] remove unused code in events ui
- [ ] how to test kafka output reliably? currently running against a local kafka docker container

Still a WIP getting a working kafka system to test against. So, I printed input to stdout and reviewed - eyeball tested the would-be output to qcr kafka. Seems to work. I'll take another swing at getting a useful dockerized kafka system running next time i'm in dr-manhattan.

### CODE REVIEW

File: client/modules/core/controllers/events.ctrl.js [hunk 1]
```diff
@@ -82,68 +81,11 @@ function EventsCtrl($scope, PostsCluster, SocialMediaPost, Event) {
     $scope.selectedEvents = events;
   }
 
-  $scope.filter = filter;
```
Comment (drJAGartner) [inline client/modules/core/controllers/events.ctrl.js]: Another added benefit of using the events as written - the UI should be more responsive!!!

File: services/dr-manhattan/event_to_kafka.py [hunk 2]
```diff
@@ -2,21 +2,23 @@
...
+from operator import itemgetter as iget
 
-def mongo_to_kafka(rec, campaign_thresh = 0.7, debug=False):
+def to_qcr_format(rec, campaign_thresh = 0.7, debug=False):
     if debug:
         print "Start conv, doing location"
         print "rec['location'] = ", rec['location']
 
-    loc = sorted(rec['location'], key=lambda x: x['weight'], reverse=True)
+    loc = sorted(rec['location'], key=iget('weight'), reverse=True)
```
Comment (drJAGartner) [inline services/dr-manhattan/event_to_kafka.py]: @lukewendling  - out of curiousity, is this supposed to be more efficient?

File: services/dr-manhattan/louvaine.py [hunk 3]
```diff
@@ -70,12 +72,12 @@ def get_text_sum(self, cluster, r_o):
                     else:
                         r_o["campaigns"]["ids"][cam] = 1
 
-            if r is not None:
-                for res in r.json():
+            if entities:
+                for res in entities:
```
Comment (drJAGartner) [inline services/dr-manhattan/louvaine.py]: Need to test this syntax with QCR kafka - there may be version specific issues with this.

File: services/dr-manhattan/event_to_kafka.py (hunk 2 above)
Comment (lukewendling) [inline services/dr-manhattan/event_to_kafka.py]: i doubt it. i saw this convenience helper when reading up on `sorted` builtin. its just a convenience for the oft-used `x -> x[n]` lambda. i wish we had this option in JS.

File: services/dr-manhattan/louvaine.py (hunk 3 above)
Comment (lukewendling) [inline services/dr-manhattan/louvaine.py]: assume u mean `json()` bits. its going thru new Loopy helpers so erthing is json object.

File: services/dr-manhattan/louvaine.py [hunk 4]
```diff
@@ -231,25 +233,17 @@ def get_communities(self):
...
+            l_tags = map(lambda x: x[0], sorted([(k, v) for k, v in d1[com]['hashtags'].iteritems()], key=iget(1)))
+            d1[com]['hashtags'] = l_tags[:10] # slice
 
-            l_terms = map(lambda x: x[0], sorted([(k, v) for k, v in d1[com]['keywords'].iteritems()], key=lambda x: x[1]))
-            if len(l_terms) > 10:
-                d1[com]['keywords'] = l_terms[:10]
-            else:
-                d1[com]['keywords'] = l_terms
+            # l_terms = map(lambda x: x[0], sorted([(k, v) for k, v in d1[com]['keywords'].iteritems()], key=lambda x: x[1]))
+            l_terms = sorted(list(d1[com]['keywords'].iteritems()), key=iget(1), reverse=1)
```
Comment (lukewendling) [inline services/dr-manhattan/louvaine.py]: @drJAGartner this wasn't reversed previously to give highest scores first. what am i missing here?

Comment (drJAGartner) [inline services/dr-manhattan/louvaine.py]: @lukewendling - probably a bug

**Comment by lukewendling** [comment]
@justinlueders ready for final review and merge

File: services/dr-manhattan/main.py [hunk 5]
```diff
@@ -11,41 +10,37 @@ def set_err(job, msg):
...
 
-
 def process_message(key,job):
     err_check(job)
     if job['state'] == 'error':
         return
 
-    host = os.getenv('HOST', job['host'])
-
+    api_root = job['api_root']
```
Comment (justinlueders) [inline services/dr-manhattan/main.py]: Why remove the Host env var?  I use that for debugging when i want to override the endpoint...
Not holding this PR up but I might want that back.

Comment (lukewendling) [inline services/dr-manhattan/main.py]: sry 'bout that. after talking to Kev abt how u guys were debugging, I was hoping to figure out a better way. let's discuss more, and of course we can bring it back if it makes life easier.

**Pull request merged**